django/db/views.py

```diff
@@ -1397,31 +1398,77 @@ def smallMoleculeDetail(request, facility_salt_id):
                 where=where, order_by=('facility_id',))        
             details['datasets'] = DataSetTable(queryset)
         
-        # nominal target dataset results information
+        # Target Affinity Spectrum data
         try:
-            dataset = DataSet.objects.get(dataset_type='Nominal Targets')
-            metaWhereClause=[
-                '''"smallMolecule" ~ '^%s' ''' % sm.facility_salt,
-                '''"isNominal" = '1' '''] 
-            ntable = DataSetManager2(dataset).get_table(
-                metaWhereClause=metaWhereClause,
-                column_exclusion_overrides=['isNominal'])
-            logger.info(str(('ntable',ntable.data, len(ntable.data))))
+            dataset = DataSet.objects.get(facility_id='20000')
+
+            sql =  '''
+                select
+                classmin,
+                array_to_string(array_agg(gene),', ') as genes
+                from (
+                select 
+                dp2.int_value as classmin,
+                dp1.text_value as gene
+                from 
+                (select dp.datarecord_id 
+                    from db_datapoint dp
+                    join db_datacolumn dc on(dp.datacolumn_id=dc.id) 
+                    join db_dataset ds on (dp.dataset_id = ds.id) 
+                    where ds.facility_id = '20000'
+                    and dc.name = 'hmsid'
+                    and dp.text_value = '%s'
+                ) as dr
+                join db_datapoint dp1 on (dp1.datarecord_id = dr.datarecord_id)
+                join db_datacolumn dc1 on (dp1.datacolumn_id = dc1.id)
+                join db_datapoint dp2 on (dp2.datarecord_id = dr.datarecord_id)
+                join db_datacolumn dc2 on (dp2.datacolumn_id = dc2.id)
+                where dc1.name = 'approvedSymbol'
+                and dc2.name = 'classmin'
+                order by classmin, gene ) a
+                group by classmin;                
+                '''            
             
-            if ntable.data: 
-                details['nominal_targets_table']=ntable
+            cursor = connection.cursor()
+            cursor.execute(sql % sm.facility_salt)
+            v = dictfetchall(cursor)
+            
+            field_lookup = {
+                1: '<b>1</b><br/>(Kd <100 nM)',
+                2: '<b>2</b><br/>(equivalent to 100 nM ≤ Kd < 1µM)',
+                3: '<b>3</b><br/>(equivalent to 1µM ≤ Kd < 10 µM)',
+                10: '<b>10</b><br/>(confirmed non-binding)'
+            }
+            
+            for tas_value in v:
+                classmin = tas_value['classmin']
+                tas_value['classmin'] = field_lookup.get(classmin, classmin)
+                
+                # Hide the non-binding genes by default (classmin==10)
+                if classmin == 10:
+                    genes = tas_value['genes']
+                    collapsible_text = \
+                        '<a class="toggle_text_collapsed" >(click here)</a>'\
```

Comment:
"Click to show" might be a better wording instead of "click here".

Also, was it considered to just do a one-way click to show and then leave it at that? There is nothing that indicates to the user that they can click to collapse the list, and I don't think it's even needed.

Reply:
True, it definitely will not be critical to re-hide it

Reply:
"click here" was Nienke's suggestion; but I don't think she had a specific reason for those words.

Reply:
If you don't need to re-hide, the jquery handler code can be dramatically simplified.